back/controllers/rangeSupervision.js

```diff
@@ -34,13 +39,56 @@ const controller = {
   },
 
   create: async function createSupervision(request, response) {
+    
+    try{
+      //fetches the supervisor id who is assgined to the range.
+      const reciever = await getUserEmail(response.req.body.supervisor);
+      //knex part that should be done in models? returns the email based on the id fetched above:
+      async function getUserEmail(key) {
+        return await knex
+          .from('user')
+          .where({ 'user.id': key })
+          .select('user.email');
+      }
+      //knex part ends.
+      var emailString = JSON.stringify(reciever);
+      //sending fetched email address to mailer.js where it is used to send message that user has been assigned a supervision
+      email('assigned', emailString);
+    } catch (error) {
+      console.error(error);
+    }
     return response
       .status(201)
       .send(response.locals.queryResult);
   },
 
   // no return here? may be a cause for a bug
   update: async function updateSupervision(request, response) {
+
+    //updates.range_supervisor returns "absent" if supervisor has not been assigned. it returns "not confirmed" is supervisor is assigned.
+    //if - checks if supervisor is assigned and only sends email if it is set. otherwise it would send email aswell when supervisor is taken off.
+    try{ 
+      const absentChecker = JSON.stringify(response.locals.updates.range_supervisor);
+      const unquoted = String(absentChecker.replace(/"([^"]+)":/g, ''));  
+      const notAbsent= '"not confirmed"';
+      if (unquoted == notAbsent){
```

Comment:
Is there a reason why you are not using

if (response.locals.updates.range_supervisor === 'not confirmed')

Reply:
All of the above could then be removed

Reply:
this method did not work when initially tried, but after taking another look clearly was not the reason for failing the sending and the problem was somewhere else. Now has been fixed and unnessesary code removed.